core/lib/zksync_core/src/dev_api_conversion_rate/mod.rs

```diff
@@ -0,0 +1,40 @@
+use axum::{extract, extract::Json, routing::get, Router};
+use tokio::sync::watch;
+use zksync_config::configs::base_token_fetcher::BaseTokenFetcherConfig;
+
+pub(crate) async fn run_server(
+    mut stop_receiver: watch::Receiver<bool>,
+    server_configs: &BaseTokenFetcherConfig,
+) -> anyhow::Result<()> {
+    let app = Router::new().route("/conversion_rate/:token_address", get(get_conversion_rate));
+
+    let bind_address = if server_configs.host.starts_with("http://") {
+        &server_configs.host[7..] // If it starts with "HTTP://", strip the prefix
+    } else {
+        &server_configs.host // Otherwise, return the original string
+    };
+
+    axum::Server::bind(&bind_address.parse().expect("Unable to parse socket address"))
+        .serve(app.into_make_service())
+        .with_graceful_shutdown(async move {
+            if stop_receiver.changed().await.is_err() {
+                tracing::warn!("Stop signal sender for conversion rate API server was dropped without sending a signal");
+            }
+            tracing::info!("Stop signal received, conversion rate server is shutting down");
+        })
+        .await
+        .expect("Conversion rate server failed");
+    tracing::info!("Conversion rate server shut down");
+    Ok(())
+}
+
+// basic handler that responds with a static string
```

Comment:
nit: to me, this comment doesn't add any clarity to the reader

Reply:
Do you think the new comment gives enough clarity? Should we add anything elese?
```rs
/// Basic handler that responds with a static string with the format expected by the token fetcher component
/// Returns 1 in the case of using ETH as the base token, a hardcoded 42 otherwise
```